BUILD

```diff
@@ -2493,6 +2495,7 @@ grpc_cc_library(
         "//src/core:grpc_backend_metric_provider",
         "//src/core:grpc_crl_provider",
         "//src/core:grpc_service_config",
+        "//src/core:grpc_tls_credentials",
```

Comment:
I don't think we want this dependency, because this will pull in the SSL libraries as a dependency, which we don't want in insecure builds.

Why is this needed?

Reply:
Good callout. To confirm, you're mostly worried including about the `libcrypto` and `libssl` external deps in the insecure build - is that correct? Or are you worried about pulling in SSL TSI as well?

If yes, I think both `libcrypto` and `libssl` are already pulled in to `grpc++_base` because they are external deps of `//src/core:grpc_crl_provider`, which is included here. However, I agree that this new dependency is pulling in SSL TSI also

The reason this is needed is because of including `src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.h` in `src/cpp/common/tls_certificate_provider.cc`. For a similar reason, we have the `//src/core:grpc_crl_provider` dependency.

One option to avoid this dependency is to add C wrapper APIs to `include/grpc/credentials.h` and use those from `src/cpp/common/tls_certificate_provider.cc`, rather than directly including `src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.h`. My understanding is that this is the older model that we were moving away from, but I may have misunderstood.

WDYT?

Reply:
Sorry, nevermind this comment -- I was confused about the BUILD target hierarchy here.  I thought I remembered that `grpc++_base` was a common target used for both secure and insecure builds, but it looks like it's actually the secure-specific target; users that want insecure builds need to use `grpc++_base_unsecure` instead.

So I think this is fine.